### PR TITLE
TILA-1095 | CSV export of applications

### PR DESCRIPTION
**Pull request by Niilis**
Add Django management command to export all applications into CSV file for manual handling.

TILA-1095

### CODE REVIEW

File: applications/utils/export_data.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    @classmethod
+    def export_application_data(cls) -> None:
+        now = timezone.now()
+        root = Path(settings.BASE_DIR)
+        file_name = f"application_data_{now.strftime('%d-%m-%Y')}.csv"
+        path = root / "exports"
+        path.mkdir(parents=True, exist_ok=True)
+
+        application_events: List[ApplicationEvent] = list(
+            ApplicationEvent.objects.all()
```
Comment (nicobav) [inline applications/utils/export_data.py:21]: I think we need to get application events per application round. So we'll need to parametrize this `ApplicationDataExporter.export_application_data` OR set this be stateful and create an instance of this and set the application_round there and then call this `export_application_data` (and change it to instance method). 
I'm perhaps more towards the parameterized classmethod call  but you can decide.

File: applications/management/commands/export_applications.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+from django.core.management.base import BaseCommand
+
+from ...utils.export_data import ApplicationDataExporter
+
+
+class Command(BaseCommand):
+    help = "Exports application data from the database"
+
+    def handle(self, *args: Any, **options: Any) -> Optional[str]:
```
Comment (nicobav) [inline applications/management/commands/export_applications.py:11]: This also should have parametrization by application round

**Review by nicobav — CHANGES_REQUESTED** [review]
Looking nice :+1: 
some tweaking with the querying the data to do :wrench:

File: applications/tests/test_application_export.py [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+
+from ..utils.export_data import ApplicationDataExporter
+from .factories import (
+    ApplicationEventFactory,
+    ApplicationEventScheduleFactory,
+    EventReservationUnitFactory,
+)
+
+
+class ApplicationExport(TestCase):
```
[the remark this answers is not in the document]
Comment (Niilis) [reply applications/tests/test_application_export.py]: Class renamed to the second suggested name.

File: applications/utils/export_data.py (hunk 1 above)
Comment (Niilis) [reply applications/utils/export_data.py]: I elected for parameterization as I believe this is the better option as well :smiley:

File: applications/management/commands/export_applications.py (hunk 2 above)
Comment (Niilis) [reply applications/management/commands/export_applications.py:19]: Parameterization implemeted to the command.

**Review by nicobav — APPROVED** [review]
Nice :+1: :rocket:

**Pull request merged**